memcached.c

```diff
@@ -4242,13 +4242,32 @@ static int enable_large_pages(void) {
     return ret;
 #elif defined(__linux__) && defined(MADV_HUGEPAGE)
     /* check if transparent hugepages is compiled into the kernel */
-    struct stat st;
-    int ret = stat("/sys/kernel/mm/transparent_hugepage/enabled", &st);
-    if (ret || !(st.st_mode & S_IFREG)) {
+    /* RH based systems possibly uses a different path */
+    static const char *mm_thp_paths[2] = {
+        "/sys/kernel/mm/transparent_hugepage/enabled",
+        "/sys/kernel/mm/redhat_transparent_hugepage/enabled",
+    };
+
+    size_t pathlen = sizeof(mm_thp_paths) / sizeof(mm_thp_paths[0]);
```

Comment:
line has a confusing smell to it. add terminating NULL/empty string and loop until empty string found instead of pathlen?

Reply:
it works here if I print out I have 2 entries.

Reply:
I believe that it works, but the common pattern in memcached code is until-null-entry. Then you also don't need the tracking variable. Or at least all the new stuff I write is.

Reply:
alright let s go for consistency.